bluetooth: fast_pair: Remove the factory reset prepare user action (#17617)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

      This configuration is now selected only by the Fast Pair use cases that require the Device Information Service (DIS).
    * The default override for the :kconfig:option:`CONFIG_BT_DIS_FW_REV_STR` Kconfig option that was set to :kconfig:option:`CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION` if :kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT` was enabled.
      The default override is now handled in the Kconfig of the Zephyr Device Information Service (DIS) module and is based on Zephyr's :ref:`zephyr:app-version-details` that uses the :file:`VERSION` file.

>     * The :c:func:`bt_fast_pair_factory_reset_user_action_prepare` weak function definition that could earlier be overridden by a user to prepare for the oncoming Fast Pair factory reset.
>       The user can still override :c:func:`bt_fast_pair_factory_reset_user_action_perform` weak function to perform custom user action during the Fast Pair factory reset.

Reviewer: ```suggestion
    * The :c:func:`bt_fast_pair_factory_reset_user_action_prepare` weak function definition, which users could previously override to prepare for the incoming Fast Pair factory reset.
      Users can still override the :c:func:`bt_fast_pair_factory_reset_user_action_perform` weak function to perform custom actions during the Fast Pair factory reset.
```

      This configuration is now selected only by the Fast Pair use cases that require the Device Information Service (DIS).
    * The default override for the :kconfig:option:`CONFIG_BT_DIS_FW_REV_STR` Kconfig option that was set to :kconfig:option:`CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION` if :kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT` was enabled.
      The default override is now handled in the Kconfig of the Zephyr Device Information Service (DIS) module and is based on Zephyr's :ref:`zephyr:app-version-details` that uses the :file:`VERSION` file.
    * The :c:func:`bt_fast_pair_factory_reset_user_action_prepare` weak function definition that could earlier be overridden by a user to prepare for the oncoming Fast Pair factory reset.

>       The function is no longer needed.

Reviewer 2: Not sure if we actually need to mention that `The function is no longer needed.` (it could depend on application's use-case). You can consider removing this sentence

Author: Removed
